src/charm.py

```diff
@@ -1216,6 +1255,16 @@ def _plugin_reconciliation(self):
             self._plugin_openid_reconciliation()
 
     def _reconciliation(self, _event):
+        if not self.state.started:
```

Comment:
It is my understanding that `self._container().can_connect()` will always return false if `not self.state.started`. Why this additional check?

Reply:
Juju limits everything that happens before the `start` event to under 40 seconds (300 seconds for the newer version of juju). If exceeded, the container agent will be marked as unhealthy and restarted by Kubernetes. Since the reconciliation process may take some time, I am deferring everything until the `start` event happens.

Reply:
Understood